future-apps/payroll/app/src/components/Table/Table.js

```diff
@@ -39,8 +58,92 @@ class Table extends React.Component {
     })
   }
 
+  paginateData = (data) => {
+    const { rowsPerPage } = this.props
+    const { currentPage } = this.state
+    const lastDataIndex = currentPage * rowsPerPage
+    const firstDataIndex = lastDataIndex - rowsPerPage
+    const paginatedData = data.slice(firstDataIndex, lastDataIndex)
+    const emptyRows = rowsPerPage - paginatedData.length
+    const totalPages = Math.ceil(data.length / rowsPerPage)
+
+    if (paginatedData.length === 0) this.setState({ currentPage: 1 })
+
+    return {
+      emptyRows,
+      paginatedData,
+      totalPages
```

Comment:
I think thi is not being used, we could remove it or use it maybe `pageNumbers ` inside `renderPageButtons`

Reply:
Yes, I'm taking it out.